src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -161,6 +162,19 @@ boolean saveAs(Path file, SaveDatabaseMode mode) {
                 .build();
         Optional<Path> selectedPath = dialogService.showFileSaveDialog(fileDialogConfiguration);
         selectedPath.ifPresent(path -> preferences.setWorkingDirectory(path.getParent()));
+        if (selectedPath.isPresent()) {
+            Path savePath = selectedPath.get();
+            // Workaround for linux systems not adding file extension
+            if (!(savePath.getFileName().toString().toLowerCase().endsWith(".bib"))) {
```

Comment:
I was beginning to use Java optionals. However, it is longer than expected.

Thus, it is OK for me to have it like it is and NOT do as follows

(not having same semantics)

```java
            if (getFileExtension(file).map(String::toLowerCase).filter(extension -> extension.equals("bib")).ifpresent()) {
```